src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -30,119 +30,85 @@
- * Suppliers should return NEW instances every time.
+ * Utility class to help with managing registry entries.<br>
+ * Maintains a list of all suppliers for entries and registers them during the proper event.<br>
+ * Suppliers should return new instances every time.
+ * <p>
+ * Example Usage:
+ * <code><pre>
+ * private static final DeferredRegister<Item> ITEMS = DeferredRegister.create(ForgeRegistries.ITEMS, MODID);
+ * private static final DeferredRegister<Block> BLOCKS = DeferredRegister.create(ForgeRegistries.BLOCKS, MODID);
  *
- *Example Usage:
- *<pre>{@code
- *   private static final DeferredRegister<Item> ITEMS = DeferredRegister.create(ForgeRegistries.ITEMS, MODID);
- *   private static final DeferredRegister<Block> BLOCKS = DeferredRegister.create(ForgeRegistries.BLOCKS, MODID);
+ * public static final DeferredHolder<Block> ROCK_BLOCK = BLOCKS.register("rock", () -> new Block(Block.Properties.create(Material.ROCK)));
+ * public static final DeferredHolder<Item> ROCK_ITEM = ITEMS.register("rock", () -> new BlockItem(ROCK_BLOCK.get(), new Item.Properties().group(ItemGroup.MISC)));
  *
- *   public static final RegistryObject<Block> ROCK_BLOCK = BLOCKS.register("rock", () -> new Block(Block.Properties.create(Material.ROCK)));
- *   public static final RegistryObject<Item> ROCK_ITEM = ITEMS.register("rock", () -> new BlockItem(ROCK_BLOCK.get(), new Item.Properties().group(ItemGroup.MISC)));
- *
- *   public ExampleMod() {
- *       ITEMS.register(FMLJavaModLoadingContext.get().getModEventBus());
- *       BLOCKS.register(FMLJavaModLoadingContext.get().getModEventBus());
- *   }
- *}</pre>
+ * public ExampleMod() {
+ *     ITEMS.register(FMLJavaModLoadingContext.get().getModEventBus());
+ *     BLOCKS.register(FMLJavaModLoadingContext.get().getModEventBus());
+ * }
+ * </pre></code>
  *
  * @param <T> The base registry type
  */
+@SuppressWarnings("deprecation") // TODO 1.20.2+ - Find+Replace RegistryObject with DeferredHolder.
 public class DeferredRegister<T>
 {
     /**
-     * DeferredRegister factory for forge registries that exist <i>before</i> this DeferredRegister is created.
+     * Factory method for DeferredRegister. The target registry does not need to exist yet.<br>
      * <p>
-     * If you have a supplier, <u>do not use this method.</u>
-     * Instead, use one of the other factories that takes in a registry key or registry name.
-     *
-     * @param reg the forge registry to wrap
-     * @param modid the namespace for all objects registered to this DeferredRegister
-     * @see #create(ResourceKey, String)
-     * @see #create(ResourceLocation, String)
+     * If the registry is never created, any created {@link DeferredHolder}s will never become {@linkplain DeferredHolder#isPresent() present}.
+     * <p>
+     * This method requires a ResourceKey for a registry - how these are obtained varies by registry.<br>
+     * For {@linkplain IForgeRegistry forge registries}, use {@link IForgeRegistry#getRegistryKey()}.<br>
```

Comment:
I suggest using an unordered list here. Additionally, this is a bad javadoc for how to get the key for a forge registry. Users using forge registries should be pointed to use the keys from `ForgeRegistries.Keys`. We should deprecate all keys that are duplicates of vanilla keys, and add javadocs to point to the vanilla key to use instead.